rockcraft/lifecycle.py

```diff
@@ -135,11 +136,17 @@ def _pack(
     base_digest: bytes,
     rock_suffix: str,
     build_for: str,
+    base_layer_dir: Path,
 ):
-    """Create the rock image for a given architecture."""
+    """Create the rock image for a given architecture.
+
+    :param base_layer_dir: The directory where the rock's base image was extracted.
+    """
     emit.progress("Creating new layer")
     new_image = project_base_image.add_layer(
-        tag=project.version, layer_path=lifecycle.prime_dir
+        tag=project.version,
+        layer_path=lifecycle.prime_dir,
+        lower_rootfs=base_layer_dir,
```

Comment:
Maybe we could use `base_layer_path` as the argument name to keep variable names consistent with `base_layer_...` and also with `layer_path`? (or `layer_dir` and `base_layer_dir`)